### ISSUE TITLE
dfind --exec doesn't handle spaces in filenames

### ISSUE THREAD

**Issue by kcgthb** [hpc/mpifileutils#311]
Filenames containing spaces are not escaped in `dfind --exec`, which leads to errors.

For instance, given the following file:
```
$ ls -al /scratch/users/kilian/test/
total 500
drwxr-xr-x  2 kilian ruthm   4096 Nov  8 07:59 .
drwx------ 24 kilian ruthm 507904 Nov  8 07:59 ..
-rw-r--r--  1 kilian ruthm      0 Nov  8 07:59 filename with spaces
```

`dfind --print` displays it fine:
```
$ srun dfind -v --type f --print  /scratch/users/kilian/test/
[2019-11-08T08:03:21] Walking /scratch/users/kilian/test
[2019-11-08T08:03:23] Walked 2 items in 1.449959 seconds (1.379349 files/sec)
/scratch/users/kilian/test/filename with spaces
```

but `dfind --exec` fails to find it:
```
$ srun dfind -v --type f --exec ls -al {} \;  /scratch/users/kilian/test/
[2019-11-08T08:04:17] Walking /scratch/users/kilian/test
[2019-11-08T08:04:19] Walked 2 items in 1.458525 seconds (1.371249 files/sec)
ls: cannot access /scratch/users/kilian/test/filename: No such file or directory
ls: cannot access with: No such file or directory
ls: cannot access spaces: No such file or directory
```

A workaround is to escape-quote the filepath placeholder `{}`:
```
# srun dfind -v --type f --exec ls -al \"{}\" \;  /scratch/users/kilian/test/
[2019-11-08T08:05:13] Walking /scratch/users/kilian/test
[2019-11-08T08:05:15] Walked 2 items in 1.506326 seconds (1.327734 files/sec)
-rw-r--r-- 1 kilian ruthm 0 Nov  8 07:59 /scratch/users/kilian/test/filename with spaces
```

but it would be nice if `dfind` could work like `find`:
```
$ find  /scratch/users/kilian/test/ -type f -exec ls -al {} \;
-rw-r--r-- 1 kilian ruthm 0 Nov  8 07:59 /scratch/users/kilian/test/filename with spaces
```

Thanks!

**Comment by adammoody** [comment]
The issue here is that the file name is copied into a string that is then passed to ```system()```.  The system call then just drops that command string into a ```bash -c ``` so that a file name with spaces generated a command like ```md5sum file with spaces```.

Here is a temporary fix that escapes space characters in file names:  https://github.com/hpc/mpifileutils/pull/326

However, now that I've done that, I don't think this is a good approach.  We probably need to be escaping other characters like tabs, newlines, and quotes.  I think the best solution would be to replace the system call with fork/exec.

**Comment by adammoody** [comment]
Here is an improved version that uses fork/exec instead of system: https://github.com/hpc/mpifileutils/pull/327

**Comment by adammoody** [comment]
Merged this PR into master, so a new build of dfind from master should now work with file names that include spaces.

**Comment by adammoody** [comment]
@kcgthb , my plan is to make a 0.10.1 bug release fix soon to pick up the fix for the MPI warnings you submitted, the fixes for the newer Lustre releases, and the dfind fixes.

I'm pretty sure this should fix the file-names-with-spaces problem for you with dfind --exec, but I'd still like to know for sure when you have a chance to test.

**Comment by adammoody** [comment]
This fix is included in v0.10.1.  Will assume this is fixed for now.
